client/app/pages/dashboards/public-dashboard-page.js

```diff
@@ -70,6 +72,38 @@ const PublicDashboardPage = {
     };
 
     this.refreshDashboard();
+
+    this.getQueryResultsErrorData = () => {
```

Comment:
Duplicate code!  😱
I didn't wanna do anything knowing #4228 is at hand.

Reply:
Considering it doesn't need to update state during the execution of the function, wdyt of moving it  to the Dashboard service? (We can later compute it with `useMemo`)

Reply:
Done  🙏